extensions/aws-sdb/deployment/src/main/java/org/apache/camel/quarkus/component/aws/sdb/deployment/AwsSdbProcessor.java

```diff
@@ -0,0 +1,68 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.aws.sdb.deployment;
+
+import com.amazonaws.auth.AWS4Signer;
+import com.amazonaws.auth.QueryStringSigner;
+import com.amazonaws.partitions.model.CredentialScope;
+import com.amazonaws.partitions.model.Endpoint;
+import com.amazonaws.partitions.model.Partition;
+import com.amazonaws.partitions.model.Partitions;
+import com.amazonaws.partitions.model.Region;
+import com.amazonaws.partitions.model.Service;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+class AwsSdbProcessor {
+
+    public static final String AWS_SDB_APPLICATION_ARCHIVE_MARKERS = "com/amazonaws";
+
+    private static final String FEATURE = "camel-aws-sdb";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
+        return new ExtensionSslNativeSupportBuildItem(FEATURE);
+    }
+
+    @BuildStep(applicationArchiveMarkers = { AWS_SDB_APPLICATION_ARCHIVE_MARKERS })
+    void process(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            BuildProducer<NativeImageResourceBuildItem> resource) {
+
+        resource.produce(new NativeImageResourceBuildItem("com/amazonaws/partitions/endpoints.json"));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false,
+                Partitions.class.getCanonicalName(),
+                Partition.class.getCanonicalName(),
+                Endpoint.class.getCanonicalName(),
+                Region.class.getCanonicalName(),
+                Service.class.getCanonicalName(),
+                CredentialScope.class.getCanonicalName(),
+                QueryStringSigner.class.getCanonicalName(),
+                AWS4Signer.class.getCanonicalName()));
```

Comment:
I was wondering for some time, which of Class.getCanonicalName() and Class.getName() is better to use here and I recently found that it is very easy to figure out: The ReflectiveClassBuildItem has constructor variants accepting java.lang.Class-es in addition to the ones accepting string class names. Internally, the Class values are transformed to strings and ta-daa ...

... the method used is Class.getName() https://github.com/quarkusio/quarkus/blob/1.3.1.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java#L38

This is just an interesting detail I wanted to share. It does not matter for non-inner classes.

Reply:
I believe we can do some cleanup in all the aws extensions in subsequent PRs, no problem for me :-)